tools/fregrid/fregrid.c

```diff
@@ -848,21 +852,29 @@ int main(int argc, char* argv[])
     if(nscalar == 0 && nvector == 0) {
       if(mpp_pe() == mpp_root_pe()) printf("NOTE from fregrid: no scalar and vector field need to be regridded.\n");
       mpp_end();
-      return 0;   
+      return 0;
     }
-    
+
     if(nscalar > 0) {
       scalar_in  = (Field_config *)malloc(ntiles_in *sizeof(Field_config));
       scalar_out = (Field_config *)malloc(ntiles_out *sizeof(Field_config));
     }
     if(nvector > 0) {
-      mpp_error("fregrid: currently does not support vertical interpolation, contact developer");
-      u_in  = (Field_config *)malloc(ntiles_in *sizeof(Field_config));
-      u_out = (Field_config *)malloc(ntiles_out *sizeof(Field_config));    
-      v_in  = (Field_config *)malloc(ntiles_in *sizeof(Field_config));
-      v_out = (Field_config *)malloc(ntiles_out *sizeof(Field_config));
+      if ((opcode & CONSERVE_ORDER1) || (opcode & CONSERVE_ORDER2  )){
```

Comment:
Last time I tried remapping u and v separately from a c192 restart file to c96 it did not work and gave me error as seen below. Are you indicating that this should work with this PR?
```
gaea11: /lustre/f2/scratch/Niki.Zadeh/tmp/tmp.ib9a2Ects1 % fregrid --input_mosaic src_grid/C192_mosaic.nc --input_file src_restart/fv_core.res --output_mosaic dst_grid/C96_mosaic.nc --output_file fregrid_c192toc96/fv_core.res --scalar_field u,v
****fregrid: first order conservative scheme will be used for regridding.
Error from pe 0: get_input_metadata(fregrid_util.c): y-size in grid file in not the same as in data file

gaea11: /lustre/f2/scratch/Niki.Zadeh/tmp/tmp.ib9a2Ects1 % fregrid --input_mosaic src_grid/C192_mosaic.nc --input_file src_restart/fv_core.res --output_mosaic dst_grid/C96_mosaic.nc --output_file fregrid_c192toc96/fv_core.res --scalar_field u
****fregrid: first order conservative scheme will be used for regridding.
Error from pe 0: get_input_metadata(fregrid_util.c): y-size in grid file in not the same as in data file

gaea11: /lustre/f2/scratch/Niki.Zadeh/tmp/tmp.ib9a2Ects1 % fregrid --input_mosaic src_grid/C192_mosaic.nc --input_file src_restart/fv_core.res --output_mosaic dst_grid/C96_mosaic.nc --output_file fregrid_c192toc96/fv_core.res --scalar_field v
****fregrid: first order conservative scheme will be used for regridding.
Error from pe 0: get_input_metadata(fregrid_util.c): x-size in grid file in not the same as in data file
```

Reply:
@nikizadehgfdl 
did the tool make_remap_file work in this case?

Reply:
@spencerkclark thanks for confirming that. I had heard of the A-grid surgery option to remap (u,v) but did not dare doing that. I think eventually someone has to do it.
@ngs333 , no, I don't know about make_remap_file. I used fregrid for all ATM restart variables except (u,v) which I took from a similar run with the target grid  hoping it adjust quickly. I used remap_land for all land restarts.